File: two-furthest-houses-with-different-colors.ts
{
    function maxDistance(colors: number[]): number {
        let max = 0;


        for (let i = 0; i < colors.length; i++) {
            const current = colors[i];
            let c = 0;

            for (let j = i + 1; j < colors.length; j++) {
                const next = colors[j];
                c += 1;

                if (next !== current && max < c)
                    max = j - i

            }

        }

        return max 
    };

    console.log(maxDistance([ 1 ,1,1, 6 ,1,1,1]))
}

File: two-furthest-houses-with-different-colors.rs
pub fn max_distance(colors: Vec<i32>) -> i32 {
    let mut result = 0;
    for idx in 0..colors.len() {
        for idx2 in idx + 1..colors.len() {
            if colors[idx] != colors[idx2] {
                result = result.max((idx as i32 - idx2 as i32).abs())
            }
        }
    }
    result
}
